rust/arrow/src/memory.rs

```diff
@@ -38,11 +38,19 @@ pub fn free_aligned(p: *mut u8, size: usize) {
 
 pub fn reallocate(ptr: *mut u8, old_size: usize, new_size: usize) -> *mut u8 {
     unsafe {
-        std::alloc::realloc(
+        let new_ptr = std::alloc::realloc(
             ptr,
             Layout::from_size_align_unchecked(old_size, ALIGNMENT),
             new_size,
-        )
+        );
+        if new_size > old_size {
+            std::ptr::write_bytes(
+                new_ptr.offset(old_size as isize),
```

Comment:
Fwiw, using the [`add`](https://doc.rust-lang.org/std/primitive.pointer.html#method.add) method is equivalent and a bit more concise:
```suggestion
                new_ptr.add(old_size),
```